src/benchmarks/gc/GC.Infrastructure/Notebooks/Reports-Impl.dib

```diff
@@ -0,0 +1,1756 @@
+        }
+
+        return info.ConfigData.Benchmarks
+            .Where(benchmark => benchmarkFilter.Include(benchmark.Key))
+            .Where(benchmark => info.Iteration < benchmark.Value.Iterations.Count)
+            .Select(benchmark => KeyValuePair.Create(benchmark.Key, benchmark.Value.Iterations[info.Iteration.Value]))
+            .Where(kvp => kvp.Value != null);
+    }
+}
+
+class TraceGCChartType : ChartType<TraceGC>
+{
+    public override BaseMetric<(string, TraceGC), XValue> DefaultXMetric { get; } = Metrics.X.GCIndex;
+    public override string DefaultBenchmarkMap(string benchmark) => benchmark;
+    
+    public override IEnumerable<SeriesInfo<TraceGC>> GetSeries(DataManager dataManager, List<Metric<TraceGC>> metrics, Filter runFilter, Filter configFilter,
+        Filter benchmarkFilter, IntFilter iterationFilter, ConfigIterationFilter configIterationFilter, IEnumerable<string> benchmarkList)
+    {
+        foreach (var metric in metrics)
+        {
+            foreach (string benchmark in benchmarkList)
+            {
+                foreach ((string run, string config, int iteration, IterationData iterationData) in
+                    dataManager.GetIterationsForBenchmark(runFilter, configFilter, iterationFilter, configIterationFilter, benchmark))
+                {
+                    yield return new (metric, run, config, null, benchmark, iteration, iterationData);
+                }
+            }
+        }
+    }
+
+    public override string GetColorFamilyKey(SeriesInfo<TraceGC> info, bool multipleMetrics, bool includeRunName, bool multipleConfigs,
+        Dictionary<string, string> configDisplayNames, bool multipleBenchmarks)
+    {
+        string benchmarkDisplay = multipleBenchmarks ? $"{info.Benchmark}, " : "";
+        string metricDisplay = multipleMetrics ? $"{info.Metric.Title}, " : "";
+        string runDisplay = includeRunName ? $"{info.Run}, " : "";
+        string configDisplay = multipleConfigs ? (configDisplayNames?.GetValueOrDefault(info.Config) ?? info.Config) : "";
+        string colorFamilyKey = $"{benchmarkDisplay}{metricDisplay}{runDisplay}{configDisplay}";
+
+        return colorFamilyKey;
+    }
+
+    public override string GetColorFamilyId(SeriesInfo<TraceGC> info, bool multipleMetrics) => $"_{info.Iteration}";
+    public override string GetSeriesTitle(SeriesInfo<TraceGC> info, string colorFamilyKey, bool multipleMetrics) => $"{colorFamilyKey}{GetColorFamilyId(info, multipleMetrics)}";
+
+    public override string GetChartTitle() => "Per-run behavior";
+
+    public override List<KeyValuePair<string, TraceGC>> GetDataSource(SeriesInfo<TraceGC> info,
+        Filter benchmarkFilter, IntFilter iterationFilter, ConfigIterationFilter configIterationFilter, Func<TraceGC, bool> dataFilter)
+        => ((IEnumerable<TraceGC>)info.IterationData.GCProcessData?.GCs).Where(gc => gc.GlobalHeapHistory != null).Where(dataFilter).Select(gc => KeyValuePair.Create("", gc));
```

Comment:
I don't know why the explicit `(IEnumerable<TraceGC>)` is required for the factored version to build.

Reply:
fixed by keeping all imports/usings together